From: Matthias Clasen <mclasen@redhat.com>
Date: Mon, 8 May 2017 08:23:01 -0400
Subject: Add a marker to rewritten desktop files

Add X-Flatpak=$app to rewritten desktop files.
Desktop files have multiple consumers, and this
makes it easier for them to know what to do.

(cherry picked from commit 66e91f55e87034208f33d10101009425a12207f6
 and e75cff6bb517badcb5c93177b4e9fe944c1e9e8d)

Origin: upstream, 0.8.6
---
 common/flatpak-dir.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/common/flatpak-dir.c b/common/flatpak-dir.c
index 924e491..557c544 100644
--- a/common/flatpak-dir.c
+++ b/common/flatpak-dir.c
@@ -3113,6 +3113,9 @@ export_desktop_file (const char   *app,
                                       "X-Flatpak-Tags",
                                       (const char * const *) tags, length);
         }
+
+      /* Add a marker so consumers can easily find out that this launches a sandbox */
+      g_key_file_set_string (keyfile, "Desktop Entry", "X-Flatpak", app);
     }
 
   groups = g_key_file_get_groups (keyfile, NULL);
